include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -659,8 +659,11 @@ struct __get_bounds_simple
         const auto& __rng1 = std::get<0>(__in_rng.tuple()); // first sequence
         const auto& __rng2 = std::get<1>(__in_rng.tuple()); // second sequence
 
-        using _SizeType = decltype(__rng1.size());
-        return std::make_tuple(_SizeType{0}, __rng1.size(), _SizeType{0}, __rng2.size());
+        using _SizeType = std::common_type_t<std::make_unsigned_t<decltype(__rng1.size())>,
+                                             std::make_unsigned_t<decltype(__rng2.size())>>;
+
+        return std::make_tuple(_SizeType{0}, _SizeType{static_cast<_SizeType>(__rng1.size())}, _SizeType{0},
```

Comment:
`_SizeType{static_cast<_SizeType>(__rng1.size())}` looks weird. Did you mean just `_SizeType{__rng1.size()}`?